### PR TITLE
Fix advertised version

### PR DESCRIPTION
**Pull request by oxarbitrage**
Closes https://github.com/zcash/zcash/issues/4375 by adding the `-` character to the list of safe ones.

Now i can see stuff like the following in the logs:

```
...
2020-04-13 14:19:37 receive version message: /MagicBean:2.1.1-1/: version 170009, blocks=795400, us=[2800:a4:316b:8e00:ceb:c7b4:3481:197f]:59754, peer=2
...
```

API call `getpeerinfo` will also gets fixed.

### CODE REVIEW

File: src/utilstrencodings.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ static const string CHARS_ALPHA_NUM = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNO
 
 static const string SAFE_CHARS[] =
 {
-    CHARS_ALPHA_NUM + " .,;_/:?@()", // SAFE_CHARS_DEFAULT
+    CHARS_ALPHA_NUM + " .,;-_/:?@()", // SAFE_CHARS_DEFAULT
```
Comment (str4d) [inline src/utilstrencodings.cpp:22]: Hmm, does this addition make sense for all the other uses of `SAFE_CHARS_DEFAULT`? Or should we add a new rule specifically for version fields?

Comment (oxarbitrage) [reply src/utilstrencodings.cpp]: I think you are right, added a new rule and force pushed.

**Comment by oxarbitrage** [comment]
What about https://github.com/zcash/zcash/pull/4444/commits/766e0b8f09c27e0a55f46581df0d8a61c51aa29c ?

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**